SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -232,6 +232,35 @@ TWeakObjectPtr<UObject> USpatialPackageMapClient::GetObjectFromUnrealObjectRef(c
 	return nullptr;
 }
 
+FNetworkGUID* USpatialPackageMapClient::GetRemovedDynamicSubobjectNetGUID(const FUnrealObjectRef& ObjectRef)
+{
+	if (FNetworkGUID* NetGUID = RemovedDynamicSubobjectObjectRefs.Find(ObjectRef))
+	{
+		return NetGUID;
+	}
+	return nullptr;
+}
+
+void USpatialPackageMapClient::AddRemovedDynamicSubobjectObjectRef(const FUnrealObjectRef& ObjectRef, const FNetworkGUID& NetGUID)
+{
+	RemovedDynamicSubobjectObjectRefs.Emplace(ObjectRef, NetGUID);
+}
+
+void USpatialPackageMapClient::ClearRemovedDynamicSubobjectObjectRefs(const Worker_EntityId& InEntityId)
+{
+	for (TMap<FUnrealObjectRef, FNetworkGUID>::TIterator dynamicSubobject(RemovedDynamicSubobjectObjectRefs); dynamicSubobject;
+		 ++dynamicSubobject)
+	{
+		UObject* Object = GetObjectFromNetGUID(dynamicSubobject->Value, true);
+		AActor* Actor = Object ? Object->GetTypedOuter<AActor>() : nullptr;
+		Worker_EntityId SubObjectEntityId = GetEntityIdFromObject(Actor);
+		if (SubObjectEntityId == InEntityId)
+		{
+			RemovedDynamicSubobjectObjectRefs.Remove(dynamicSubobject->Key);
```

Comment:
I'm pretty sure removing the key from the map will invalidate the iterator, need to do this instead:
```suggestion
			DynamicSubobjectIterator.RemoveCurrent();
```